src/adapter/src/catalog.rs

```diff
@@ -5192,6 +5235,65 @@ impl Catalog {
                 Op::DropTimeline(timeline) => {
                     tx.remove_timestamp(timeline);
                 }
+                Op::GrantRole {
+                    role_id,
+                    member_id,
+                    grantor_id,
+                } => {
+                    let existing_role = state.get_role_mut(&member_id);
+                    if is_reserved_name(&existing_role.name) {
+                        return Err(AdapterError::Catalog(Error::new(
+                            ErrorKind::ReservedRoleName(existing_role.name.clone()),
+                        )));
+                    }
```

Comment:
Should this be caught higher in the stack? If the role already exists then what's the point of failing here?

Reply:
Got it, that totally makes sense!

> My initial thought is to error on this too.

Without a ton of context on RBAC or what roles like `mz_system` can do, I would agree that we should error in the `GRANT mz_system to <role>` case too. IMO it's better to be more restrictive up front and later relax those restrictions, than trying to go the other way

Reply:
Good point, I've added this second error case.